gubernator/static/build.js

```diff
@@ -99,13 +100,54 @@ function expand_all(btn) {
 	btn.remove();
 }
 
+/* given a string containing ansi formatting directives, return a new one
+   with designated regions of text marked with the appropriate color directives,
+   and with all unknown directives stripped */
+function ansi_to_html(orig) {
+	function annotate(p1, p2, string) {
+		var p1n = +(p1.replace('0;', ''));
+		if (p1n == 0) // reset
+			return p2;
+		else if (p1n === 1) // bold
+			return '<em>' + p2 + '</em>';
+		else if (30 <= p1n && p1n <= 37) // foreground color
+			return '<span class="ansi-' + (p1n - 30) + '">' + p2 + '</span>'
+		else if (90 <= p1n && p1n <= 97) // foreground color, bright
+			return '<span class="ansi-' + (p1n - 90 + 8) + '">' + p2 + '</span>'
+		return string;
+	}
+	// Find color codes, optionally followed by a bold code.
+	var filtered = orig.replace(/\033\[([0-9;]*)\w(\033\[1m)?([^\033]*?)\033\[0m/g, function(match, p1, p2, p3, offset, string) {
```

Comment:
This is a tricky regular expression. I would appreciate at the very least a longer comment saying what the pieces are. Even better would be a test :)